src/Logging/Logging/LogService.cs

```diff
@@ -1,64 +1,54 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace CodeMonkeys.Logging
 {
-    internal sealed class LogServiceComposition : ILogService
+    internal sealed class LogService : ILogService
     {
-        private readonly ContextAwareLogServiceProvider[] _providers;
+        private readonly IScopedLogService[] _services;
 
-        public LogServiceComposition(ContextAwareLogServiceProvider[] providers)
+        public bool IsEnabled { get; set; } = true;
+
+        public LogService(IScopedLogService[] services)
         {
-            _providers = providers;
+            _services = services;
         }
 
-        public bool IsEnabledFor(LogLevel logLevel)
+        public void EnableScopedService<TScopedLogService>()
+            where TScopedLogService : IScopedLogService
         {
-            List<Exception> exceptions = null;
-
-            foreach (var provider in _providers)
-            {
-                var service = provider.LogService;
-
-                try
-                {
-                    if (!service.IsEnabledFor(logLevel))
-                        continue;
-
-                    return true;
-                }
-                catch (Exception e)
-                {
-                    if (exceptions == null)
-                        exceptions = new List<Exception>();
+            var serviceType = typeof(TScopedLogService);
+            var service = _services.FirstOrDefault(s => s.GetType() == serviceType);
 
-                    exceptions.Add(e);
-                }
-            }
+            service.DisableLogging();
+        }
 
-            if (exceptions != null && exceptions.Count > 0)
-                throw new AggregateException(
-                    "Error(s) occured while querying the associated log services!",
-                    exceptions);
+        public void DisableScopedService<TService>() 
+            where TService : IScopedLogService
+        {
+            var serviceType = typeof(TService);
+            var service = _services.FirstOrDefault(s => s.GetType() == serviceType);
 
-            return false;
+            service.DisableLogging();
         }
 
         public void Log<TState>(
-            DateTimeOffset timestamp, 
+            DateTimeOffset timestamp,
             LogLevel logLevel, 
             TState state, 
             Exception ex, 
             Func<TState, Exception, string> formatter)
         {
+            if (!IsEnabled)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5626772)